### PR TITLE
Fix segmentation bug with excluded unary ops

### LINKED ISSUE #1541: Simple slice fusion fails to segment
Minimal repro:
```python
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id1(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.BFloat16)
    T1 = fd.ops.neg(T0)
    T2 = fd.ops.slice(T1, start_indices=[0], end_indices=[5])
    fd.add_output(T2)

with FusionDefinition() as fd:
    nvfuser_fusion_id1(fd)

inputs = [
    torch.randn((10,), dtype=torch.bfloat16, device='cuda:0')
]
fd.execute(inputs)
```
This fails with `Can not find a scheduler to schedule fusion segment`. `NVFUSER_DUMP=segmenter_logging` shows:
```
***Runtime***: Try to schedule fusion un-segmented:

Scheduler _no_op_ ***rejected*** because : output has a concrete dimension
Scheduler _reduction_ ***rejected*** because : No reduction op to schedule
Scheduler _transpose_ ***rejected*** because : First input of SliceOp must be fusion input.
Scheduler _pointwise_ ***rejected*** because : First input of SliceOp must be fusion input.
Scheduler _inner_persistent_ ***rejected*** because : needs a reduction op
Scheduler _outer_persistent_ ***rejected*** because : needs a reduction op
Scheduler _inner_outer_persistent_ ***rejected*** because : needs a reduction op
Scheduler _matmul_ ***rejected*** because : Matmul scheduler supports fusions only with a single MMA op, got: 0

**Segmenter** Considering fusion:
i13 = fmin(i0, 5);
i15 = fmax(0, i13);
T2_g[ iS4{( i0 + ( ( fmax(0, ( fmin(i0, 5) )) ) + ( -i0 ) ) )}rf ]
   = slice( T1_g[ iS1{i0} ], { {0, i15, 1} } )

Scheduler _no_op_ ***rejected*** because : output has a concrete dimension
Scheduler _reduction_ ***rejected*** because : No reduction op to schedule
Scheduler _transpose_ ***rejected*** because : cannot find two mismatching inner most dimensions
Scheduler _transpose_ ***rejected*** because : cannot find two mismatching inner most dimensions
***Accepted*** as: pointwise

**Segmenter** Considering fusion:
T1_g[ iS1{i0} ]
   = -T0_g[ iS0{i0} ];
i13 = fmin(i0, 5);
i15 = fmax(0, i13);
T2_g[ iS4{( i0 + ( ( fmax(0, ( fmin(i0, 5) )) ) + ( -i0 ) ) )}rf ]
   = slice( T1_g[ iS1{i0} ], { {0, i15, 1} } )

Scheduler _no_op_ ***rejected*** because : output has a concrete dimension
Scheduler _reduction_ ***rejected*** because : No reduction op to schedule
Scheduler _transpose_ ***rejected*** because : First input of SliceOp must be fusion input.
Scheduler _pointwise_ ***rejected*** because : First input of SliceOp must be fusion input.
Scheduler _inner_persistent_ ***rejected*** because : needs a reduction op
Scheduler _outer_persistent_ ***rejected*** because : needs a reduction op
Scheduler _inner_outer_persistent_ ***rejected*** because : needs a reduction op
Scheduler _matmul_ ***rejected*** because : Matmul scheduler supports fusions only with a single MMA op, got: 0
An error occurred while executing nvFuser FusionDefinition 1.
```

### PR DESCRIPTION
**Pull request by naoyam**
Fixes #1541

### CODE REVIEW

**Comment by naoyam** [comment]
!build --diff --diff-bench

**Comment by naoyam** [comment]
!build --diff

**Comment by naoyam** [comment]
!build --diff --diff-bench
